packages/components/src/components/SquiggleChart.tsx

```diff
@@ -1,10 +1,117 @@
 import * as React from "react";
 import _ from "lodash";
-import { run, errorValueToString } from "@quri/squiggle-lang";
+import styled from "styled-components";
+import {
+  run,
+  errorValueToString,
+  squiggleExpression,
+} from "@quri/squiggle-lang";
 import type { samplingParams, exportEnv } from "@quri/squiggle-lang";
 import { NumberShower } from "./NumberShower";
 import { DistributionChart } from "./DistributionChart";
 import { ErrorBox } from "./ErrorBox";
+import useSize from "@react-hook/size";
+
+const variableBox = {
+  Component: styled.div`
+    background: white;
+    border: 1px solid #eee;
+    border-radius: 2px;
+    margin-bottom: 0.4em;
+  `,
+  Heading: styled.div`
+    border-bottom: 1px solid #eee;
+    padding-left: 0.8em;
+    padding-right: 0.8em;
+    padding-top: 0.1em;
+  `,
+  Body: styled.div`
+    padding: 0.4em 0.8em;
+  `,
+};
+
+export const VariableBox: React.FC<{
+  heading: string;
+  children: React.ReactNode;
+}> = ({ heading = "Error", children }) => {
+  return (
+    <variableBox.Component>
+      <variableBox.Heading>
+        <h3>{heading}</h3>
+      </variableBox.Heading>
+      <variableBox.Body>{children}</variableBox.Body>
+    </variableBox.Component>
+  );
+};
+
+export interface SquiggleItemProps {
+  /** The input string for squiggle */
+  expression: squiggleExpression;
+  width: number;
+  height: number;
+}
+
+const SquiggleItem: React.FC<SquiggleItemProps> = ({
+  expression,
+  width,
+  height,
+}: SquiggleItemProps) => {
+  switch (expression.tag) {
+    case "number":
+      return (
+        <VariableBox heading="Number">
+          <NumberShower precision={3} number={expression.value} />
+        </VariableBox>
+      );
+    case "distribution": {
+      let distType = expression.value.type();
+      return (
+        <VariableBox heading={`Distribution (${distType})`}>
+          {distType === "Symbolic" ? (
+            <>
+              <div>{expression.value.toString()}</div>
+            </>
+          ) : (
+            <></>
+          )}
+          <DistributionChart
```

Comment:
It would be nice to separate out DistributionChart and have separate stories for that. (The continuous/discrete/mixed ones should be moved there). However, that would be more work, can address later.